antlr-kotlin-target/src/main/resources/org/antlr/v4/tool/templates/codegen/Kotlin/Kotlin.stg

```diff
@@ -215,7 +215,7 @@ public open class <file.grammarName>BaseVisitor\<T> : AbstractParseTreeVisitor\<
      * The default implementation returns the result of calling [visitChildren] on [ctx].
      */
     override fun visit<lname; format="cap">(ctx: <file.parserName>.<lname; format="cap">Context): T {
-        return this.visitChildren(ctx)!!
+        return visitChildren(ctx)!!
```

Comment:
I personally find the `this.` useful in the template, as it removes ambiguity. In the template one cannot use the IDE to resolve the method, and so having a little bit more information helps

Reply:
The issue with `this` is it must be consisentent. Or we use it everywhere, or nowhere.  
Currently it was being used on half the cases.

The template is generally edited once in a while, we know all variables refer to the parser state, as local variables are always prefixed with `_`.

Reply:
> The issue with this is it must be consistent. Or we use it everywhere, or nowhere.
Currently it was being used on half the cases.

Then I would think it would make sense to move toward using it everywhere (not as part of this PR)

> The template is generally edited once in a while, we know all variables refer to the parser state, as local variables are always prefixed with _.

That is not immediately obvious to someone looking to this file for the first time or from time to time

Reply:
> Then I would think it would make sense to move toward using it everywhere

IMO it's counterproductive. Using it everywhere bloats the source code so much it becomes unreadable, especially on big generated files.

Reply:
I'll test with the C++ grammar and if it looks decent I'll create another PR to add the this qualifier to all the instance state.